Assets/HoloToolkit/Input/Scripts/Interactable.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Base class Interactable GameObjects can inherit from.
+    /// </summary>
+    public class Interactable : MonoBehaviour
+    {
+        protected bool TriggerKeywordOnGazeOnly;
+
+        protected bool IsFocused;
+
+        protected virtual void OnEnable()
+        {
+            GestureManager.Instance.OnTap += OnTap;
+            GazeManager.Instance.OnGazeExit += OnGazeExit;
+            GazeManager.Instance.OnGazeEnter += OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized += KeywordRecognized;
+        }
+
+        protected virtual void OnDisable()
+        {
+            GestureManager.Instance.OnTap -= OnTap;
+            GazeManager.Instance.OnGazeExit -= OnGazeExit;
+            GazeManager.Instance.OnGazeEnter -= OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized -= KeywordRecognized;
+        }
+
+        /// <summary>
+        /// Called when a user has tapped any gameObject.
+        /// </summary>
+        /// <param name="tappedGameObject">GameObject user has tapped.</param>
+        protected virtual void OnTap(GameObject tappedGameObject)
+        {
+            if (tappedGameObject == gameObject)
```

Comment:
First, I agree 100% that the SendMessage() call is bad.  It's bad for performance since it uses reflection.  It encourages sloppy coding.   It makes scripts less portable.  Getting rid of SendMessage is a good idea.  

However there must be an intersection point N where SendMessage and signaling N events has the same performance.  As N increases SendMessage is cheaper.  As N decreases signaling the events is cheaper.  This number N exists.   I might try to discover 'N' in a simple application.  The article you linked tested up to N=5.  It isn't comparing C# events with SendMessage, so the side by side comparison in the article isn't relevant to my point (I see how it is for the keywordmanager wireup, but this is orthogonal).

If N turns out to be a relatively small number (less than 100?) then we should look at alternatives and an interface is an alternative (which we should, of course, measure the performance impact). I'm not sure that I consider 'avoid' to be the same as a 'No no'.  :)